app/storage/storage.py

```diff
@@ -0,0 +1,89 @@
+from abc import abstractmethod, ABC
+from functools import cached_property
+
+from flask import current_app
+
+from app.data_model import app_models
+
+
+class StorageModel:
+    TABLE_CONFIG = {
+        app_models.SubmittedResponse: {
+            "key_field": "tx_id",
+            "expiry_field": "expires_at",
+            "table_name_key": "EQ_SUBMITTED_RESPONSES_TABLE_NAME",
+            "schema": app_models.SubmittedResponseSchema,
+        },
+        app_models.QuestionnaireState: {
+            "key_field": "user_id",
+            "table_name_key": "EQ_QUESTIONNAIRE_STATE_TABLE_NAME",
+            "schema": app_models.QuestionnaireStateSchema,
+        },
+        app_models.EQSession: {
+            "key_field": "eq_session_id",
+            "expiry_field": "expires_at",
+            "table_name_key": "EQ_SESSION_TABLE_NAME",
+            "schema": app_models.EQSessionSchema,
+        },
+        app_models.UsedJtiClaim: {
+            "key_field": "jti_claim",
+            "expiry_field": "expires_at",
+            "table_name_key": "EQ_USED_JTI_CLAIM_TABLE_NAME",
+            "schema": app_models.UsedJtiClaimSchema,
+        },
+    }
+
+    def __init__(self, model=None, model_type=None):
+        self._model = model
+        self._model_type = type(model) if model else model_type
+
+        if not self._model_type:
+            raise ValueError("One of model/model_type is required")
+
+        if self._model_type not in self.TABLE_CONFIG:
+            raise KeyError("Invalid model_type provided")
+
+        self._config = self.TABLE_CONFIG[self._model_type]
+
+    @cached_property
+    def schema(self):
+        return self._config["schema"]()
+
+    @cached_property
+    def key_field(self):
+        return self._config["key_field"]
+
+    @cached_property
+    def expiry_field(self):
+        return self._config.get("expiry_field")
+
+    @cached_property
+    def item(self):
+        if self._model:
+            return self.schema.dump(self._model)
```

Comment:
I don't think this is very clear. The model is only used on the load, so this seems more of a serialise method. There is also an inconsistency as the deserialisation for the `get` is done in the implementations. Maybe we should introduce `serialise` and `deserialise` methods here?

Reply:
`StorageModel` now handles serialization and deserialization.